### PR TITLE
Yarn: Add tests for Yarn workspaces

### PR DESCRIPTION
**Pull request by fviernau**
Workspace support is not fully implemented yet.
Add a test for illustrating the current behavior and to easily illustrate
the effect of changes in implemented behavior.

### CODE REVIEW

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces-expected-output.yml [hunk 1]
```diff
@@ -0,0 +1,181 @@
+---
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces-expected-output.yml]: This file belongs to the next commit.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "workspaces": [
+      "packages/*"
+  ],
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "private": true,
+  "keywords": [],
+  "author": "",
+  "license": "ISC"
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json]: License should be Apache-2.0 to match our own license.
Comments on this file apply to the two `package.json`s below as well.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "main": "index.js",
+  "workspaces": [
+      "packages/*"
+  ],
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "private": true,
+  "keywords": [],
+  "author": "",
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json]: Remove empty "keywords" and "author", they are not required.
I think "scripts" and "main" can be removed as well for consistency with our NPM test projects. Except if they are required for Yarn workspaces to work, of course.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "workspace",
+  "version": "1.0.0",
+  "description": "",
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json]: Please add a description similar to the one we use in the NPM test projects.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    spdx_expression: "ISC"
+  vcs:
+    type: ""
+    url: ""
+    revision: ""
+    path: ""
+  vcs_processed:
+    type: "Git"
+    url: "ssh://git@github.com/heremaps/oss-review-toolkit.git"
+    revision: "04847ae49d9bb93810a2a83ce70c8a2aba71ceec"
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml]: Use `<REPLACE_URL>` and `<REPLACE_REVISION>`.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces-expected-output.yml (hunk 1 above)
Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces-expected-output.yml]: Done.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json (hunk 2 above)
Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json]: Done.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json (hunk 3 above)
Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json]: Done.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json (hunk 4 above)
Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json]: Done.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml (hunk 5 above)
Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml]: Done.

Comment (mnonnenmacher) [reply analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml]: Values are still hardcoded, also in the packages below.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/packages/pkg1/index.js [hunk 6]
```diff
@@ -0,0 +1 @@
+console.log("hello");
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/packages/pkg1/index.js]: Can this file be removed?

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml [hunk 7]
```diff
@@ -0,0 +1,181 @@
+---
+project:
+  id: "Yarn::workspace:1.0.0"
+  definition_file_path: "analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/package.json"
+  declared_licenses:
+  - "ISC"
+  declared_licenses_processed:
+    spdx_expression: "ISC"
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml]: This needs to be updated to Apache-2.0 now as well.

Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml]: Done.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/packages/pkg1/index.js (hunk 6 above)
Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/packages/pkg1/index.js]: Done.

File: analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml (hunk 5 above)
Comment (fviernau) [reply analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces/yarn-workspaces-expected-output.yml]: Done.

**Pull request merged**